cli/internal/cmd/upgradeapply_test.go

```diff
@@ -83,6 +83,15 @@ func TestUpgradeApply(t *testing.T) {
 			wantErr:           true,
 			stdin:             "no\n",
 		},
+		"abort, rollback terraform err": {
+			kubeUpgrader: &stubKubernetesUpgrader{
+				currentConfig: config.DefaultForAzureSEVSNP(),
```

Comment:
can we check that TF is restored when the upgrade is aborted (use the mock for that?) and also not called when we don't abort?

Reply:
I don't have a strong opinion on mocks, but I think we should discuss if we want to actively use mock testing in the team and either do it in every package when someone edits it

Reply:
I agree that mocking is a last resort. Of course it would be more resilient to only mock the filesystem and actually check that the files are restored

Reply:
I do see the point in using mocks for verifying that the expected code path is taken here. However, I don't think that this fits very well with our current testing strategy here, as it would require us to encode logic in our tests, which is often recognized as an anti-pattern. (see [here](https://abseil.io/resources/swe-book/html/ch12.html), 12-14 to 12-16) I do see the values coming from these tests, but I think if we decide to adopt this, we should do so as a whole. I think we should avoid mocking only in certain tests, to keep some consistency. I might have a wrong understanding of how to use these mocks too. If you see a way to use it without putting too much logic into the test, feel free to let me know.